### PR TITLE
Make composites blog post

### PR DESCRIPTION
**Pull request by eveilyeverafter**
short post describing how to make composite images using a python script.

### CODE REVIEW

File: posts/composite-images/making-composite-images.qmd [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+      - ref: nstg
+      - ref: mrvikramkohli
+toc: true
+toc-title: Contents
+toc-depth: 3
+toc-expand: 2
+toc-location: left
+number-sections: true
+number-depth: 4
+date: "2024-06-11"
```
Comment (crwilliams11) [inline posts/composite-images/making-composite-images.qmd]: date bump

File: posts/composite-images/making-composite-images.qmd [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+format-links: [pdf, docx]
+---
+
+```{r}
+#| eval: true
+#| echo: false
+#| label: "fig-composite"
+#| fig-width: 4
+#| fig-height: 4
+#| fig-cap: "A composite image created using the make_composity.py script.
```
Comment (crwilliams11) [inline posts/composite-images/making-composite-images.qmd]: composity -> composite

File: posts/composite-images/making-composite-images.qmd [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+```{r}
+#| eval: true
+#| echo: false
+#| label: "fig-composite"
+#| fig-width: 4
+#| fig-height: 4
+#| fig-cap: "A composite image created using the make_composity.py script. 
+#|  This image represents all channels of a single field of view in the 
+#|  [publicly available](https://nanostring.com/products/cosmx-spatial-molecular-imager/ffpe-dataset/cosmx-smi-mouse-brain-ffpe-dataset/){target='blank'} 
+#|  mouse coronal hemisphere FFPE."
```
Comment (crwilliams11) [inline posts/composite-images/making-composite-images.qmd]: FFPE **dataset**.

File: posts/composite-images/making-composite-images.qmd [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+
+- Variable: `compress_value` (set to 3). Lossless file compression value. Higher values produce smaller files at 
+the expense of increased script execution time. The set value is a 
+compromise between file size and execution time.
+
+# Output
+
+- `raw` – The extracted tif files from the morphology 2D 
+images will be saved in this folder. 
+The file format will follow <fov_num>_ch<#>_raw.<user_format>. <fov_num> is the fov number,`ch<#>` is the channel number (from 0 to 4), and <user_format> is the
```
Comment (crwilliams11) [inline posts/composite-images/making-composite-images.qmd]: The formatting doesn't appear to be working on rendering: 
image
I think the "<>" are  not getting interpreted as desired.

File: posts/composite-images/making-composite-images.qmd [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+  Example
+  ```
+  F001_ch0_raw.jpg (for user_format = jpg)
+  ```
+  
+- `8bit` - The images in the raw_folder 
+are converted to 8bit and saved in this folder. Note: If the specified 
+user_format is jpg, the 8bit files are identical to the raw files. 
+The file format will follow <fov_num>_ch<#>_8bit.<user_format> except
```
Comment (crwilliams11) [inline posts/composite-images/making-composite-images.qmd]: Formatting issue on render: 
image

File: posts/composite-images/making-composite-images.qmd [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+
+  Example
+  ```
+  F001_ch0_8bit.tif (for user_format = tif)
+  F001_ch0_raw.jpg (for user_format = jpg)
+  ```
+
+- `8bit_autocontrast` – Images in the 8bit folder are autocontrasted 
+based on the user supplied clipping value. The file format will follow 
+fov_num>_ch<#>_8bit_autocontrast.<user_format>
```
Comment (crwilliams11) [inline posts/composite-images/making-composite-images.qmd]: formatting issue, similar to above

File: posts/composite-images/making-composite-images.qmd [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+
+  Example
+  ```
+  F001_ch0_8bit_autocontrast.png (for user_format = png)
+  ```
+
+- `composite` - Composite images created from the images in the 
+`8bit` folder. The composite type is a screen composite. 
+The file format will follow fov_num>_composite.<user_format>
```
Comment (crwilliams11) [inline posts/composite-images/making-composite-images.qmd]: formatting issue similar to above

File: posts/composite-images/making-composite-images.qmd [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+
+  Example
+  ```
+  F001_composite.jpg (for user_format = jpg)
+  ```
+
+- `composite_autocontrast` -  Images in the composite_folder are 
+autocontrasted based on the user specified clipping value. 
+The file format will follow fov_num>_composite_autocontrast.<user_format>
```
Comment (crwilliams11) [inline posts/composite-images/making-composite-images.qmd]: formatting issue similar to above

File: posts/composite-images/making-composite-images.qmd [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+
+  Example
+  ```
+  F001_composite_autocontrast.png (for user_format = png)
+  ```
+
+# Usage
+
+```
+navigate to your Morphology2D folder
```
Comment (crwilliams11) [inline posts/composite-images/making-composite-images.qmd]: This appears as code, so may be informative to start this comment with "#"

File: posts/composite-images/making-composite-images.qmd [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+```
+
+Regex pattern matching on 2D morphology file name format is implemented, 
+however, only NanoString 2D morphology files should be present in the 
+folder containing the make_composite script. 
+
+# Example
+
+The example dataset that we used was the mouse coronal hemisphere
+FFPE that is available to download from NanoString's website [here](https://nanostring.com/products/cosmx-spatial-molecular-imager/ffpe-dataset/cosmx-smi-mouse-brain-ffpe-dataset/){target="blank"}.
```
Comment (crwilliams11) [inline posts/composite-images/making-composite-images.qmd]: FFPE **dataset**

File: posts/composite-images/making-composite-images.qmd [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+```{default filename="Terminal"}
+#| eval: false
+python /path/to/CosMx-Analysis-Scratch-Space/assets/make-composite/make_composite.py
+```
+
+```{r}
+#| eval: true
+#| echo: false
+#| label: "fig-run"
+#| fig-cap: "Screenshot of standard output from terminal following script execution. In this example, I set the clipping percentage to 3 and the output to `png`. On an Macbook Pro M1, this took about 25 minutes to process 130 FOVs."
```
Comment (crwilliams11) [inline posts/composite-images/making-composite-images.qmd]: On **a** Macbook...

File: posts/composite-images/making-composite-images.qmd [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+...
+├── 20230406_205644_S1_C902_P99_N99_F129.TIF
+├── 20230406_205644_S1_C902_P99_N99_F130.TIF
+├── 8bit
+├── 8bit_autocontrast
+├── composite
+├── composite_autocontrast
+└── raw
+```
+
```
Comment (crwilliams11) [inline posts/composite-images/making-composite-images.qmd:241]: Can you add a concluding sentence about what I should do next? A sentence or two, similar to this: These composite images can now be imported into open-source software or explored further.

**Review by crwilliams11 — COMMENTED** [review]
Minor suggestions. Main comment is that some of the formatting is not rendering correctly for file  name structures, see individual comments for example images. Additionally, a concluding sentence or two would help wrap it up.

File: posts/composite-images/making-composite-images.qmd (hunk 9 above)
Comment (eveilyeverafter) [reply posts/composite-images/making-composite-images.qmd]: generally navigate to folder. I will make it `cd`.

**Review by crwilliams11 — APPROVED** [review]
All changes look good. Approved.

**Pull request merged**